cachito/workers/pkg_managers/ruby.py

```diff
@@ -0,0 +1,69 @@
+from pathlib import Path
+
+import packagedcode.gemfile_lock as gemlock_parser
+
+from cachito.errors import ValidationError
+
+
+class GemMetadata:
+    """Gem metadata."""
+
+    def __init__(self, name, version, gem_type, path):
+        """Fill GemMetadata with data."""
+        self.name = name
+        self.version = version
+        self.type = gem_type
+        self.path = path
+
+    def __eq__(self, other):
+        if isinstance(other, GemMetadata):
+            return (
+                self.name == other.name
+                and self.version == other.version
+                and self.type == other.type
+                and self.path == other.path
+            )
+        return False
+
+
+def parse_gemlock(gemlock_path: Path):
+    """Parse dependencies from Gemfile.lock.
+
+    :param Path gemlock_path: the full path to Gemfile.lock
+    :return: list of Gems
+    """
+    dt = gemlock_parser.GemfileLockParser(str(gemlock_path)).dependency_tree
+    dependencies = []
+
+    for key in dt:
+        gem = dt[key]
+        validate_gem_metadata(gem, gemlock_path.parents[0])
+        dependencies.append(GemMetadata(gem.name, gem.version, gem.type, gem.remote))
+
+    return dependencies
+
+
+def validate_gem_metadata(gem, gemlock_dir):
+    """Validate parsed Gem.
+
+    :param Gem gem: gem with information parsed from Gemfile.lock
+    :param Path gemlock_dir: the root directory containing Gemfile.lock
+    :raise: ValidationError
+    """
+    if gem.name is None or gem.version is None:
+        raise ValidationError("Unspecified name or version of a RubyGem.")
+
+    if gem.type == "GEM":
+        if gem.remote != "https://rubygems.org/":
+            raise ValidationError(
+                "Cachito supports only https://rubygems.org/ as a remote for Ruby GEM dependencies."
+            )
+    elif gem.type == "GIT":
+        if not gem.remote.startswith("https://"):
+            raise ValidationError("All Ruby GIT dependencies have to use HTTPS protocol.")
```

Comment:
Could you remind me why ssh:// is not allowed?

Reply:
From design doc
> Maintaining credentials is a major increase to the scope of this design

and I don't know about any product we want to support that would be using SSH for git repos.

From your comment, I understand that credentials are somehow already managed, so if you really wanted to see this feature in Cachito, I could probably implement it as well, but I haven't looked into it yet, so I don't know how complex it would be.

Reply:
AFAIK no additional changes are required, the Red Hat deployment of Cachito has an ssh key configured for git so cloning via ssh should "just work". At least for public repos - for private ones, only github is supported via the [RH-Cachito](https://github.com/RH-Cachito) user.

It's OK to disallow ssh urls until the need arises, but I think Cachito can support them (to the extent above) for free